scene/main/scene_tree.cpp

```diff
@@ -135,6 +145,9 @@ SceneTree::Group *SceneTree::add_to_group(const StringName &p_group, Node *p_nod
 	E->get().nodes.push_back(p_node);
 	//E->get().last_tree_version=0;
 	E->get().changed = true;
+
+	node_grouped(p_node, p_group);
```

Comment:
The only change that I would do is emit the signal right here, instead to call this function.

Reply:
I added the node_grouped and node_ungrouped methods because I noticed the existing `node_added`, `node_removed` signals were handled the same way, and while I personally don't see a great need for them, I wanted to maintain code consistency.

If you believe they're not necessary, I can easily remove them and emit the signal directly, as you suggested.

Reply:
Make sense, thanks.